scipy/stats/_rcont/_rcont.c

```diff
@@ -0,0 +1,242 @@
+/**
+  This file contains two algorithms written in C to generate random two-way
+  tables. The algorithms rcont1 and rcont2 originate from Boyett and Patefield,
+  respectively. For more information, see the docs for each function.
+
+  If you wonder about the spelling, rcont is short for random contingency
+  table. Random contingency table is a bit of a misnomer. The tables generated
+  by these algorithms have no contingency/association between the two
+  variables.
+
+  Author: Hans Dembinski
+*/
+
+#include "_rcont.h"
+#include <math.h>
+
+// helper function to access a 1D array like a C-style 2D array
+double *ptr(double *m, int nr, int nc, int ir, int ic)
+{
+  return m + nc * ir + ic;
+}
+
+// sanity checks for inputs to rcont1 and rcont2,
+// also sets total number of entries
+int rcont_check(double *n, const double *m, int nr, const double *r, int nc, const double *c)
```

Comment:
This function is unused. Also, the parameter `m` isn't used anywhere in the function (other than just checking if it is a non-null pointer). Did you forget to call this in the `rcont1` and `rcont2` functions? I think the Python function `_process_parameters` does all these tests. Not sure if this is needed at all.

Reply:
The function was removed.